training/distributed_training/pytorch/model_parallel_v2/llama_v2/smp-finetuning-llama-fsdp-tp.ipynb

```diff
@@ -0,0 +1,959 @@
+    "- `train_lib.py`: This has functions for running an end-to-end training of the GPT-NeoX or Llama-v2 model with SMP FSDP, settings for hybrid sharding applied, and implemented with code lines to save, load, and fine-tune the model.\n",
+    "- `train_utils.py`: This has utility functions for training.\n",
+    "\n",
+    "## Additional Resources\n",
+    "- To learn more about launching a multi-node distributed PyTorch training job, see [Launching a Distributed Training Job](https://sagemaker.readthedocs.io/en/stable/frameworks/pytorch/using_pytorch.html#launching-a-distributed-training-job).\n",
+    "- To learn more about using the SageMaker Python SDK with PyTorch, see [Using PyTorch with the SageMaker Python SDK](https://sagemaker.readthedocs.io/en/stable/frameworks/pytorch/using_pytorch.html).\n",
+    "- To learn more about launching a training job in Amazon SageMaker with your own training image, see [Use Your Own Training Algorithms](https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-training-algo.html).\n",
+    "\n",
+    "## Prerequisites\n",
+    "You need to create an S3 bucket to store the input data for training. This bucket must be located in the same AWS Region that you choose to launch your training job. To learn how to create a S3 bucket, see (Create your first S3 bucket in the Amazon S3 documentation)[https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html].\n",
+    "\n",
+    "## Launching Environment\n",
+    "### Amazon SageMaker Notebook\n",
+    "You can run the notebook on an Amazon SageMaker notebook instance without manually setting your aws credentials.\n",
+    "1. Create a new SageMaker notebook instance and open it.\n",
+    "2. Zip the contents of this folder & upload to the instance with the Upload button on the top-right.\n",
+    "3. Open a new terminal with `New -> Terminal`.\n",
+    "4. Within the terminal, enter the correct directory and unzip the file.\n",
+    "    1. `cd SageMaker && unzip <your-zip-name-here>.zip`\n",
+    "\n",
+    "### Locally\n",
+    "You can run locally by launching a jupyter notebook server with `jupyter notebook`. This requires you to set your aws credentials in the environment manually. See [Configure the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) for more details."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Amazon SageMaker Initialization\n",
+    "Run the following cell to import SageMaker modules and retrieve information of your current SageMaker work environment, such as your AWS account ID, the AWS Region, and the ARN of your Amazon SageMaker execution role. Upgrade SageMaker SDK to the latest version.\n",
+    "\n",
+    "**NOTE:** This step might require a kernel restart."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "CONTAINER_IMAGE = \"...\"\n",
```

Comment:
I think we should specify why these variables are required, or add them to the relevant cells, e.g fsx creation cell for fsx variables

Reply:
It's intentional to consolidate those vars, as internal vs external values are different.
- Easier for cleanup for the github repo.